mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -231,6 +240,85 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         except Exception as err:
             raise MlflowException(err)
 
+    def _azure_adls_gen2_upload_file(self, credentials, local_file, artifact_path):
+        """
+        Uploads a file to a given Azure storage location using the ADLS gen2 API.
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials.headers)
+
+            # try to create the file
+            try:
+                put_adls_file_creation(credentials.signed_uri, headers=headers)
+            except requests.HTTPError as e:
+                if e.response.status_code in [401, 403]:
+                    _logger.info(
+                        "Failed to authorize ADLS file creation request, possibly due "
+                        "to credential expiration. Refreshing credentials and trying again..."
+                    )
+                    credential_info = self._get_write_credential_infos(
+                        run_id=self.run_id, paths=[artifact_path]
+                    )[0]
+                    put_adls_file_creation(credential_info.signed_uri, headers=headers)
+                else:
+                    raise e
+
+            # next try to patch the file
+            offset = 0
+            is_first_chunk = True
+            use_single_part_upload = False
+            for chunk in yield_file_in_chunks(local_file, _AZURE_MAX_BLOCK_CHUNK_SIZE):
+                try:
+                    cur_offset = len(chunk)
+                    if is_first_chunk and cur_offset < _AZURE_MAX_BLOCK_CHUNK_SIZE:
+                        use_single_part_upload = True
+                    patch_adls_file_upload(
+                        credentials.signed_uri,
+                        chunk,
+                        offset,
+                        headers=headers,
+                        is_single=use_single_part_upload,
+                    )
+                    is_first_chunk = False
+                    offset += cur_offset
+                except requests.HTTPError as e:
+                    if e.response.status_code in [401, 403]:
+                        _logger.info(
+                            "Failed to authorize ADLS patch request, possibly due to "
+                            "credential expiration. Refreshing credentials and trying again..."
+                        )
+                        credential_info = self._get_write_credential_infos(
+                            run_id=self.run_id, paths=[artifact_path]
+                        )[0]
+                        patch_adls_file_upload(
+                            credential_info.signed_uri,
```

Comment:
Shouldn't `credentials` be reset as `credentials_infos` here before retrying? What happens for the next chunk?

Reply:
It seems that all the occurrences of `credentials_infos` in this function should be replaced with `credentials`.

Reply:
Will fix.